addons/point_of_sale/models/stock_picking.py

```diff
@@ -29,19 +29,20 @@ def _create_picking_from_pos_order_lines(self, location_dest_id, lines, picking_
             location_id = picking_type.default_location_src_id.id
             positive_picking = self.env['stock.picking'].create({
                 'partner_id': partner.id if partner else False,
-                'user_id': False,
+                'user_id': self.env.user.id,
                 'picking_type_id': picking_type.id,
                 'move_type': 'direct',
                 'location_id': location_id,
                 'location_dest_id': location_dest_id,
             })
 
             positive_picking._create_move_from_pos_order_lines(positive_lines)
-            try:
-                with self.env.cr.savepoint():
-                    positive_picking._action_done()
-            except (UserError, ValidationError):
-                pass
+            if force_done:
+                try:
+                    with self.env.cr.savepoint():
+                        positive_picking._action_done()
+                except (UserError, ValidationError):
+                    pass
```

Comment:
You should probably call the 'generate_activity_for_pending_pickings' in the except here, like that it will work in all cases, the one with closing at the end of session, and the one when it is done by_order

Reply:
@pimodoo changes have done except this one, because we need a bunch of picking like this when "in real time" option is selected.

```
Some operations could not be confirmed due to exceptions :

  - WH/OUT/XXXX
  - WH/OUT/YYYY
```

Reply:
Thnaks, I'll have a look